### DISCUSSION TITLE
Upgrading Vue Router independently from Quasar v1

### DISCUSSION THREAD

**Discussion by jonfelixrico** [quasarframework/quasar#16051] [General - Components / Directives / etc] [answered]
Hello, I would like to upgrade Vue Router separately from Quasar. Will there be any risk of breaking the library components if I do it?

For context, the project that I'm working on is using `@quasar/app@1.7.4` and `vue-router@3.1.6`. I would like to upgrade vue-router to version 3.5.0.

I've did some reading of the Vue Router changelogs from 3.1.6 to 3.5.0 and [changes to the `router-link`](https://github.com/vuejs/vue-router/blob/dev/CHANGELOG.md#350-2021-01-25) caught my eye since I'm guessing that Quasar could've been using them internally which could cause some problems (just guessing)

Responses are much appreciated!

**Comment by yusufkandemir** [comment, marked as answer]
It's entirely under your control when using Quasar v2. It's different in v1, which is in control of Quasar. You will need to use `resolutions`/`overrides` fields in `package.json` depending on your package manager to override the version properly. But, you will most likely run into issues, both because of the changes you've mentioned and some other stuff. There is an old PR about upgrading vue-router here, which explains some of the problems: #14295

**Reply by jonfelixrico** [reply]
I see... so there's a risk of running into Issues. That's unfortunate. I guess the safest option is to upgrade Quasar to a minor version that has the Vue Router that I'm looking for.

Anyways, thank you so much for answering, I appreciate it!
